cranelift/codegen/src/isa/unwind/systemv.rs

```diff
@@ -22,6 +22,77 @@ pub enum RegisterMappingError {
     UnsupportedRegisterBank(&'static str),
 }
 
+fn into_raw_expr(expr: gimli::write::Expression) -> Expression {
+    use gimli::read::UnwindSection;
+    use gimli::{read, write, Encoding, Format, Register, RunTimeEndian};
+
+    pub struct Writer {
+        writer: write::EndianVec<RunTimeEndian>,
+    }
+    impl write::Writer for Writer {
+        type Endian = RunTimeEndian;
+        fn endian(&self) -> Self::Endian {
+            self.writer.endian()
+        }
+        fn len(&self) -> usize {
+            self.writer.len()
+        }
+        fn write(&mut self, bytes: &[u8]) -> write::Result<()> {
+            self.writer.write(bytes)
+        }
+        fn write_at(&mut self, offset: usize, bytes: &[u8]) -> write::Result<()> {
+            self.writer.write_at(offset, bytes)
+        }
+    }
```

Comment:
This looks like it is purely delegating to `EndianVec<RunTimeEndian>`, which already implements `Writer`, so why are we wrapping it, instead of using it directly?

Reply:
Nice to know, thanks! I removed this portion of the code in favor of not dealing with `Expression` serialization at all.